### PR TITLE
(feature-wip)[parquet-reader] support page index serde

### PR DESCRIPTION
**Pull request by wsjz**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [x] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/format/parquet/vparquet_reader.h [hunk 1]
```diff
@@ -48,40 +44,46 @@ namespace doris::vectorized {
...
+    void _fill_block_data(std::vector<tparquet::ColumnChunk> columns);
+    bool _has_page_index(std::vector<tparquet::ColumnChunk> columns);
+    Status _process_page_index(std::vector<tparquet::ColumnChunk> columns);
 
 private:
     FileReader* _file_reader;
     std::shared_ptr<FileMetaData> _file_metadata;
-    //    const int64_t _batch_size;
-    //    const int32_t _num_of_columns_from_file;
+    std::shared_ptr<RowGroupReader> _row_group_reader;
```
Comment (morningman) [inline be/src/vec/exec/format/parquet/vparquet_reader.h:75]: In the new design, we should ensure that the row group reader is a reader that can be processed independently in parallel.

File: be/src/vec/exec/format/parquet/vparquet_group_reader.cpp [hunk 2]
```diff
@@ -17,4 +17,63 @@
 
 #include "vparquet_group_reader.h"
 
-namespace doris::vectorized {} // namespace doris::vectorized
\ No newline at end of file
+namespace doris::vectorized {
+
+RowGroupReader::RowGroupReader(doris::FileReader* file_reader,
+                               std::shared_ptr<FileMetaData> file_metadata,
+                               std::vector<int> column_ids)
```
Comment (morningman) [inline be/src/vec/exec/format/parquet/vparquet_group_reader.cpp:24]:
```suggestion
                               const std::vector<int>& column_ids)
```

File: be/src/vec/exec/format/parquet/vparquet_group_reader.cpp [hunk 3]
```diff
@@ -17,4 +17,63 @@
...
+    //    }
+}
+
+Status RowGroupReader::read_next_row_group(const int32_t* group_id) {
+    int32_t total_group = _file_metadata->num_row_groups();
+    while (_current_row_group < total_group) {
+        bool filter_group = false;
+        _process_row_group_filter(&filter_group);
+        if (!filter_group) {
+            group_id = &_current_row_group;
```
[the remark this answers is not in the document]
Comment (wsjz) [reply be/src/vec/exec/format/parquet/vparquet_group_reader.cpp:45]: It is a template implement this version

File: be/src/vec/exec/format/parquet/vparquet_reader.h (hunk 1 above)
Comment (wsjz) [reply be/src/vec/exec/format/parquet/vparquet_reader.h:75]: stay implement after this version

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**